### ISSUE TITLE
Preserve neighbor velocity in NeighborEncoder or document why it is disabled

### ISSUE THREAD

**Issue by Max-Bin** [tier4/Diffusion-Planner#193]
## Problem

`NeighborEncoder.forward()` explicitly zeros the neighbor velocity channels before encoding:

```python
x = torch.cat([x[..., :4], torch.zeros_like(x[..., 4:6]), x[..., 6:]], dim=-1)
```

The input layout is documented as:

```python
(x, y, cos, sin, vx, vy, w, l, type(3))
```

so this removes `vx` and `vy` from the neighbor history token before the temporal/channel encoder sees it.

## Impact

The neighbor encoder cannot directly use tracked-object velocity, even though the data contains it. It can only infer motion from the retained position history, which is further limited by the recent-history truncation in the encoder input.

This is a potential information-loss issue for:

- moving-neighbor intent and speed estimation;
- collision-related scene understanding;
- sparse or partially backfilled histories, where position deltas may be less reliable than the explicit velocity signal.

If this was intentional, the reason should be documented. If not, the velocity channels should be preserved and the behavior should be covered by a regression test.

## Expected behavior

Either:

1. keep neighbor `vx, vy` in the encoder input; or
2. document why velocity must be disabled and make this behavior explicit/configurable.
